### PR TITLE
Added ArgMax operator

### LINKED ISSUE #14: Implement Argmax operator
ONNX operator: https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ArgMax-13

### LINKED ISSUE #210: Update golangci-lint version

### PR DESCRIPTION
**Pull request by Swopper050**
Closes #14
Closes #210

### CODE REVIEW

**Comment by Swopper050** [comment]
@wipsel What should we do about the string constant lint error?

The attributes 'axis' and 'keepdims' occur in (at least) 3 operators, causing the linter to say it should be a constant. However, it does not really feel nice to group some of the attributes together and some not (like `select_last_index` in this operator). Do you have an idea how we could do this?

File: Makefile [hunk 1]
```diff
@@ -58,7 +58,7 @@ install: ## Install project with its depedencies.
 
 install_lint: ## Install the linter.
 	curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh \
-		| sh -s -- -b $(shell go env GOPATH)/bin v1.50.1
+		| sh -s -- -b $(shell go env GOPATH)/bin v1.57.2
```
Comment (wipsel) [inline Makefile]: Maybe it is better to upgrade the linter in a different issue?

**Comment by wipsel** [comment]
> @wipsel What should we do about the string constant lint error?
> 
> The attributes 'axis' and 'keepdims' occur in (at least) 3 operators, causing the linter to say it should be a constant. However, it does not really feel nice to group some of the attributes together and some not (like `select_last_index` in this operator). Do you have an idea how we could do this?

I see two options. Make all attribute names a constant in their operator. That would make the linter happy and i don't really see a problem with it. It could be a nice way to add documentation about the attribute if need be. 

Another option would be to create an `attrs` or `attrname` package defining the attribute names of all operators "globally" 
The usage would than look something like this: 

```go

// in attrs/const.go
const (
    KeepDims = "keep_dims"
    Axis = "axis"
)

// in some operator 
import attrs


// Init initializes the argmax operator.
func (a *ArgMax) Init(n *onnx.NodeProto) error {
	attributes := n.GetAttribute()
	for _, attr := range attributes {
		switch attr.GetName() {
		case attrs.Axis:
                     // do  attr stuff
               }
       } 
}
```

I don't know how much of the attr names are shared and if it justifies a seperate package. Maybe you have a better view on this. If not you could define them under their own name under their own operator. I would however make it consistent with all the attr names for the same operator. So if you change them to a const for this i would include the `select_last_index` as wel. Don't know if it all makes sense but hope it helps

**Comment by Swopper050** [comment]
> I see two options. Make all attribute names a constant in their operator. That would make the linter happy and i don't really see a problem with it. It could be a nice way to add documentation about the attribute if need be.

Let's do this. I created an issue #211 for it and for this MR only fixed it in ArgMax.

**Review by wipsel — APPROVED** [review]

**Pull request merged**